Standards/scs-0100-w1-flavor-naming-implementation-testing.md

```diff
@@ -0,0 +1,69 @@
+---
+title: "SCS Flavor Naming Standard: Implementation and Testing Notes"
+type: Supplement
+track: IaaS
+status: Proposal
+supplements:
+  - scs-0100-v1-flavor-naming.md
+  - scs-0100-v2-flavor-naming.md
+  - scs-0100-v3-flavor-naming.md
+---
+
+## Introduction
+
+The three major versions of the standard that exist so far are very similar, and deliberately so.
+Therefore, the procedures needed to implement or test them are very similar as well. Yet, this document
+will only cover v3, because v1 and v2 are already obsolete by the time of writing.
+
+## Implementation Notes
+
+Every flavor whose name starts with `scs-` must conform with the naming scheme laid down in the standard.
+
+### Operational Tooling
+
+#### Syntax Check
+
+The [test suite](../Tests/iaas/flavor-naming) comes with a handy
+[command-line utility](../Tests/iaas/flavor-naming/cli.py) that can be used to validate flavor names, to
+interactively construct a flavor name via a questionnaire, and to generate prose descriptions for given
+flavor names. See the [README](../Tests/iaas/flavor-naming/README.md) for more details.
+
+The functionality of this script is also (partially) exposed via the web page
+<https://flavors.scs.community/>.
+
+With the OpenStack tooling (`python3-openstackclient`, `OS_CLOUD`) in place, you can call
+`cli.py -v parse v3 $(openstack flavor list -f value -c Name)` to get a report
+on the syntax compliance of the flavor names of the cloud environment.
+
+#### Flavor Creation
+
+The [OpenStack Flavor Manager](https://github.com/osism/openstack-flavor-manager) will create a whole set
+of flavors in one go, given a YAML description of this set.
+
+## Automated Tests
+
+### Errors
+
+The following items MUST be detected and reported as an error:
+
+- any syntax error in a name starting with `scs-`,
+- any mismatch between any immediately discoverable property of a flavor (currently, CPU, RAM and disk size)
+  and the meaning of its name (which is usually a lower bound), such as the CPU generation or hypervisor.
+
+In addition, the following items MAY be reported as an error:
+
+- any mismatch between any non-immediately discoverable property of flavor and the meaning of its name.
+
+### Warnings
+
+None so far.
+
+### Implementation
+
+The script [`flavor-names-openstack.py`](../Tests/iaas/flavor-naming/flavor-names-openstack.py)
```

Comment:
this should be absolute: `https://github.com/SovereignCloudStack/standards/blob/main/Tests/iaas/flavor-naming/flavor-names-openstack.py`

Reply:
The relative version works very well, both locally and on Github.
The version `/Tests/iaas/flavor-naming/flavor-names-openstack.py` is absolute, for it begins with a slash, and therefore it is not objected to by Markdownlint, and it also works very well with Github, but the Link checker doesn't like it.
The version `https://github.com/SovereignCloudStack/standards/blob/main/Tests/iaas/flavor-naming/flavor-names-openstack.py` is the one that I personally like least. It doesn't work locally, and it's hard to see that it refers to something from the same repository.
Can we find a way to get this to work without a full URL including a host?

Reply:
Links starting with slash such as  `/Tests/iaas/flavor-naming/flavor-names-openstack.py` are root-relative URLs. Without slash like `Tests/iaas/flavor-naming/flavor-names-openstack.py` are _true_ relative links. 

I totally agree that the absolute solution `https://github.com/SovereignCloudStack/standards/blob/main/Tests/iaas/flavor-naming/flavor-names-openstack.py` or even the true perma absolute `https://github.com/SovereignCloudStack/standards/blob/772e06b1707609cd0602a24d10b1bd3e027782c8/Tests/iaas/flavor-naming/flavor-names-openstack.py` are the least attractive.

But with `https://github.com/SovereignCloudStack/standards` repository being implemented within the `docs.scs.community` where there is a link-checker aswell, it's the solution that works best without implementing another build script. It would be possible with another build script that is processing all links within all .md files being cloned from each repository and replacing them with absolute links that are defined within the `docs.package.json`.

Reply:
Fair point. Thanks for the clarification.